.github/workflows/analysis-of-endpoint-connections.yml

```diff
@@ -15,48 +9,77 @@ env:
   java: 21
 
 jobs:
-  analysis-of-endpoint-connections:
+  parse-rest-calls-and-endpoints:
     timeout-minutes: 10
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout code
+      - name: checkout code
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Get list of modified files
-        run: |
-          git diff --name-only origin/${{ github.event.pull_request.base.ref }} HEAD > modified_files.txt
+      - name: set up JDK 21
+        uses: actions/setup-java@v2
+        with:
+          java-version: '${{ env.java }}'
+          distribution: 'adopt'
 
-      # Analyze the client sided REST-API calls
-      - name: Set up Node.js
+      - name: set up node.js
         uses: actions/setup-node@v4
         with:
           node-version: '${{ env.node }}'
 
-      - name: Install and compile TypeScript
+      - name: parse client sided REST-API calls
         run: |
-          cd supporting_scripts/analysis-of-endpoint-connections/src/main/typeScript/
           npm install
-          tsc -p tsconfig.analysisOfEndpointConnections.json
-
-      - name: Run analysis-of-endpoint-connections-client
-        run: |
+          tsc -p supporting_scripts/analysis-of-endpoint-connections/src/main/typeScript/tsconfig.analysisOfEndpointConnections.json
           node supporting_scripts/analysis-of-endpoint-connections/src/main/typeScript/AnalysisOfEndpointConnectionsClient.js
 
-      - name: Upload JSON file
+      - name: parse server sided Endpoints
+        run: ./gradlew :supporting_scripts:analysis-of-endpoint-connections:runEndpointParser
+
+      - name: upload parsing results
         uses: actions/upload-artifact@v4
         with:
-          name: rest-calls-json
-          path: supporting_scripts/analysis-of-endpoint-connections/restCalls.json
+          name: parsing results
+          path: |
+            supporting_scripts/analysis-of-endpoint-connections/endpoints.json
+            supporting_scripts/analysis-of-endpoint-connections/restCalls.json
+
+  analysis-of-endpoint-connections:
+    needs: parse-rest-calls-and-endpoints
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
 
-      # Analyze the server sided endpoints
-      - name: Set up JDK 21
+      - name: set up JDK 21
```

Comment:
Why make them lowercase? The custom steps in the github workflow usually start with a capital letter

![image](https://github.com/user-attachments/assets/eedbc701-4402-4d67-8cc8-e335f178cb8a)